pennylane/beta/plugins/default_tensor.py

```diff
@@ -191,86 +173,168 @@ def _add_node(self, A, wires, name="UnnamedNode", key="state"):
 
         return node
 
-    def _add_edge(self, node1, idx1, node2, idx2):
-        """Adds an edge between two nodes.
+    def _add_initial_state_nodes(self, tensors, wires, names):
+        """Create the nodes representing the initial input state circuit.
 
-        Args:
-            node1 (tn.Node): first node to connect
-            idx1 (int): index of node1 to connect the edge to
-            node2 (tn.Node): second node to connect
-            idx2 (int): index of node2 to connect the edge to
+           Input states can be factorized or entangled. If a state can be factorized
+           into :math:`k` subsystems, then ``tensors``, ``wires``, and ``names`` should be
+           sequences of length :math:`k`.
+
+           ``self._free_wire_edges`` is updated with the dangling edges from the prepared state nodes.
+
+           If ``self._rep == "mps"``, then the ``self.mps`` attribute is replaced with a new
+           matrix product state object representing the prepared initial states.
+
+          Args:
+              tensors (Sequence[np.array, tf.Tensor, torch.Tensor]): the numerical tensors for each
+               factorized component of the state (in the computational basis)
+              wires (Sequence(list[int])): the wires for each factorized component of the state
+              names (Sequence[str]): name for each factorized component of the state
+        """
+        if not (len(tensors) == len(wires) == len(names)):
+            raise ValueError("tensors, wires, and names must all be the same length.")
 
+        if self._rep == "exact":
+            self._free_wire_edges = []
+            for tensor, wires_seq, name in zip(tensors, wires, names):
+                if len(tensor.shape) != len(wires_seq):
+                    raise ValueError(
+                        "Tensor provided has shape={}, which is incompatible "
+                        "with provided sequence of wires {}.".format(tensor.shape, wires_seq)
+                    )
+                node = self._add_node(tensor, wires=wires_seq, name=name)
+                self._free_wire_edges.extend(node.edges)
+
+        elif self._rep == "mps":
+            nodes = []
+            for tensor, wires_seq, name in zip(tensors, wires, names):
+                if len(tensor.shape) != len(wires_seq):
+                    raise ValueError(
+                        "Tensor provided has shape={}, which is incompatible "
+                        "with provided sequence of wires {}.".format(tensor.shape, wires_seq)
+                    )
+                tensor = self._expand_dims(tensor, [0, -1])
+                if tensor.shape == (1, 2, 1):
```

Comment:
Is it worth checking for other shapes as well here? E.g. `(2, 1, 1)` could be fine as well would just need reshaping.

Reply:
The key here is really the 2. The 1s are added internally by this class itself, so they should have some promise of being in the correct spot (i.e., these won't be supplied by user)